src/network/http/websocket.cpp

```diff
@@ -625,11 +627,35 @@ namespace fc { namespace http {
    }
    void websocket_server::listen( const fc::ip::endpoint& ep )
    {
-      my->_server.listen( boost::asio::ip::tcp::endpoint( boost::asio::ip::address_v4(uint32_t(ep.get_address())),ep.port()) );
+       my->_server.listen( boost::asio::ip::tcp::endpoint( boost::asio::ip::address_v4(uint32_t(ep.get_address())),ep.port()) );
+   }
+
+   uint16_t websocket_server::get_listening_port()
+   {
+       websocketpp::lib::asio::error_code ec;
+       return my->_server.get_local_endpoint(ec).port();
    }
 
    void websocket_server::start_accept() {
-      my->_server.start_accept();
+       my->_server.start_accept();
+   }
+
+   void websocket_server::stop_listening()
+   {
+       my->_server.stop_listening();
+   }
+
+   void websocket_server::close()
+   {
+       for (auto& connection : my->_connections)
+           my->_server.close(connection.first, websocketpp::close::status::normal, "Goodbye");
+   }
+
+   void websocket_server::synchronous_close()
+   {
+      close();
+      if (my->_closed)
```

Comment:
I don't think this works as intended. `my->_closed` is empty until the destructor of `my` is called, which doesn't happen until the websocket_server itself is destroyed.

Reply:
You are definitely correct; I'll have `_closed` get populated when the first connection is established.

Either way, it doesn't work as intended, though, since it's intended to allow destroying the `websocket_server` without a race condition, but as far as I can tell, websocketpp gives no means of doing this without putting the server on its own thread, which we do not currently do. So the `synchronous_close` method is pointless and misleading: they do not do what they say they do.

I can just leave it be, drop the `synchronous_close` method entirely, or rewrite it so the server is on its own thread. What do you think?

Reply:
Agree to drop `synchronous_close` for now.